### PR TITLE
Bugfix/issue 138 request validation

### LINKED ISSUE #138: Request validation using validation libaray (zod, yup, joi)
**Description:**

Implement request validation for all request inputs using a validation library (Zod, Yup, Joi). This ensures that all incoming data is properly validated before processing.

**Tasks:**

- Choose a validation library (Zod, Yup, Joi).
- Implement validation schemas for all request inputs.
- Integrate the validation schemas into the request handling logic.
- Add error handling for validation failures.

**Acceptance Criteria:**

- All request inputs are validated using the chosen validation library.
- Validation errors are appropriately handled and returned to the client.
- Comprehensive unit tests are written to ensure validation is working as expected.

### PR DESCRIPTION
**Pull request by dsmabulage**
## Purpose

The purpose of this PR is to fix #138 

## Goals
Add request validations via schema library to maintain consistent API parameters.

## Approach
Add Zod lib to schema validation
validation is done in the middleware level

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/middlewares/requestValidator.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        return res
+          .status(400)
+          .json({ error: 'Invalid data', details: errorMessages })
+      }
+      return res.status(500).json({ error: 'Internal server error' })
+    }
+  }
+}
+
+export const requestQueryValidator = (schema: ZodSchema<any, any, any>) => {
```
Comment (anjula-sack) [inline src/middlewares/requestValidator.ts:23]: Can we define the types here?

File: src/middlewares/requestValidator.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  return (req: any, res: any, next: any) => {
```
Comment (anjula-sack) [inline src/middlewares/requestValidator.ts:24]: Here as well. Let's avoid using any type

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/middlewares/requestValidator.ts (hunk 1 above)
Comment (dsmabulage) [reply src/middlewares/requestValidator.ts]: resolved

File: src/middlewares/requestValidator.ts (hunk 2 above)
Comment (dsmabulage) [reply src/middlewares/requestValidator.ts]: resolved

**Comment by dileepainivossl** [comment]
@anjula-sack comments resolved

**Pull request merged**